src/Glpi/Http/LegacyPostRequestActionsListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Glpi\Http;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class LegacyPostRequestActionsListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::FINISH_REQUEST => ['onFinishRequest'],
+        ];
+    }
+
+    public function onFinishRequest(): void
+    {
+        $this->resetSessionAjaxParam();
+        $this->triggerGlobalsDeprecation();
+    }
+
+    private function resetSessionAjaxParam(): void
+    {
+        \Session::resetAjaxParam();
+    }
+
+    private function triggerGlobalsDeprecation(): void
+    {
+        global $AJAX_INCLUDE;
+        if (isset($AJAX_INCLUDE)) {
+            \Toolbox::deprecated('Using the global "$AJAX_INCLUDE" variable has been removed. Use "$this->setAjax()" from your controllers instead.', version: "11.0");
+        }
+
+        global $SECURITY_STRATEGY;
+        if (isset($SECURITY_STRATEGY)) {
+            \Toolbox::deprecated('Using the global "$SECURITY_STRATEGY" variable has been removed. Use proper Route attributes in your controllers instead.', version: "11.0");
+        }
```

Comment:
See my comment on the `inc/includes.php` file. Triggering deprecations here will not be detected by PHPStan. We should do this in the `inc/includes.php` file and maybe also here ONLY if the `inc/includes.php` file is not part of the `get_included_files()` result. Also, in this case, we should trigger warnings or exceptions, as we could not prevent BC-breaks.